lib/chef/provider/package/zypper.rb

```diff
@@ -139,7 +157,7 @@ def zypper_version
         end
 
         def zypper_package(command, global_options, *options, names, versions)
-          zipped_names = zip(names, versions)
+          zipped_names = source_files_exist? ? [new_resource.source] : zip(names, versions)
```

Comment:
i think you can drop array-ifying the source property, but then you need a check to fix the zypper_version < 1.0 case:

```
          zipped_names = new_resource.source ? new_resource.source : zip(names, versions)
          names = zipped_names if new_resource.source
```